compile/plugin.py

```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/env python3
```

Comment:
This might be a bit hard to understand without context. Can you add some explanation about this plug in and the standard protoc generator in the PR description? Is it simply a slimmed down version of protoc or does it do anything special?